rclcpp/test/rclcpp/test_memory_strategy.cpp

```diff
@@ -307,6 +307,8 @@ TEST_F(TestMemoryStrategy, get_node_by_group) {
     EXPECT_EQ(
       node_handle,
       memory_strategy()->get_node_by_group(callback_group, weak_groups_to_nodes));
+    // Clear the handles to not hold NodeBase.
+    memory_strategy()->clear_handles();
```

Comment:
is this line needed or can we delete it without modifying anything else?
If we cannot delete it, could you explain what changes?

Sorry for making all these questions, but there are many objects with interdependent lifetimes and it's hard to follow why this addition is needed.

Reply:
> is this line needed or can we delete it without modifying anything else?
> If we cannot delete it, could you explain what changes?

This line can be removed, but I need to update 
https://github.com/ros2/rclcpp/blob/0b890eea1d83e6ff2bc6ca8df7f192b1151e6599/rclcpp/test/rclcpp/test_memory_strategy.cpp#L314-L316 to expect getting the callback_group as the NodeBase(AllocatorMemoryStrategy-> VectorRebind<std::shared_ptr<const rcl_subscription_t>> -> NodeBase, as we know, some subscriptions(e.g. /parameter_events) and services are created when a Node created) is still alive even if the Node goes out of scope.

What I did is try to not update the expectation of a test before. I can remove that line and update the expectation. But there might be a new class derived from the `memory_strategy::MemoryStrategy` and not using the `std::shared_ptr<const rcl_subscriptions_t>` to keep the entity in the future, the expectation could be failed. So I'd like to make use of calling this virtual member function `MemoryStrategy::clear_handles()` to clean `NodeBase` holder, and then to expect a `nullptr` in the test.

What do you think about it?

Reply:
I see, I think the change looks good.